test/functional/wallet_signer.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test external signer.
+
+Verify that a bitcoind node can use an external signer command
+"""
+import os
+import platform
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_rpc_error,
+)
+
+
+class SignerTest(BitcoinTestFramework):
+    def mock_signer_path(self):
+        path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'mocks', 'signer.py')
+        if platform.system() == "Windows":
+            return "py " + path
+        else:
+            return path
+
+    def set_test_params(self):
+        self.num_nodes = 3
+
+        self.extra_args = [
+            [],
+            ['-signer=%s' % self.mock_signer_path()],
+            ['-signer=%s' % "fake.py"],
```

Comment:
In 2926553d26e2f15e7986a73f6ee678112ab8ab7f "test: add external signer test"

Use f-strings

```suggestion
            [f"-signer={self.mock_signer_path()}"],
            ["-signer=fake.py"],
```

Reply:
Ah, we upgraded to Python 3.6 "recently" :-)

Reply:
It's funny but also a bit shameful that this has taken so long that the drift of linters and python versions is starting to eat into it.

Reply:
Oh I've been plowing through rebases while descriptor wallets were being built. This is nothing :-)